docker_mcp/services/stack.py

```
@@ -396,66 +449,501 @@
+                            backup_info = {"success": False, "backup_path": None}
+                            
+                except Exception as e:
+                    self.logger.warning("Backup creation failed", error=str(e), stack=stack_name, target=target_host_id)
+                    migration_steps.append(f"⚠️  Backup failed: {str(e)} - continuing with migration (RISKY)")
             
             # Step 6: Transfer archive to target
-            if all_paths and not dry_run:
+            if all_paths and not dry_run and archive_path:
                 migration_steps.append("🚀 Transferring data to target host...")
+                temp_suffix = os.urandom(8).hex()[:8]
+                target_archive_path = f"/tmp/{stack_name}_migration_{temp_suffix}.tar.gz"
```

Ruff (0.12.2)

Probable insecure usage of temporary file or directory: "/tmp/"

(S108)